backend/fms_core/services/index.py

```diff
@@ -0,0 +1,99 @@
+from fms_core.models import Index, IndexSet, IndexStructure, Sequence, SequenceByIndex3Prime, SequenceByIndex5Prime
+
+from django.core.exceptions import ValidationError
+
+def get_or_create_index_set(set_name):
+    index_set = None
+    created_entity = False
+    errors = []
+    warnings = []
+
+    if set_name:
+        try:
+            index_set = IndexSet.objects.get(name=set_name)
+
+        except IndexSet.DoesNotExist:
+            try:
+                index_set = IndexSet.objects.create(name=set_name)
+                created_entity = True
+            # Pile up all validation error raised during the creation of the container
+            except ValidationError as e:
+                errors.append(';'.join(e.messages))
+    else:
+        errors.append(f"The name of the set is required to get or create a set.")
+
+    return (index_set, created_entity, errors, warnings)
+
+def create_index(index_set, index_name, index_structure):
+    index = None
+    errors = []
+    warnings = []
+
+    index_structure_obj = None
+    if index_structure:
+        try:
+            index_structure_obj = IndexStructure.objects.get(name=index_structure)
+        except IndexStructure.DoesNotExist:
+            errors.append(f"Invalid index structure.")
+            return index, errors, warnings
+
+        try:
+            index = Index.objects.create(index_set=index_set, name=index_name, index_structure=index_structure_obj)
+        except ValidationError as e:
+            errors.append(';'.join(e.messages))
+
+        return index, errors, warnings
+
+
+    else:
+        errors.append(f"Index structure is needed to create an index.")
```

Comment:
This else statement has no return ... this would have a default return of None but this function typically returns more than one value so it's likely to cause failure if the else is called.